keras/src/layers/attention/multi_head_attention.py

```diff
@@ -415,9 +422,28 @@ def _compute_attention(
           attention_output: Multi-headed outputs of attention computation.
           attention_scores: Multi-headed attention weights.
         """
-        # Note: Applying scalar multiply at the smaller end of einsum improves
-        # XLA performance, but may introduce slight numeric differences in
-        # the Transformer attention head.
+        if config.is_flash_attention_enabled() and return_attention_scores:
+            raise ValueError(
+                "Returning attention scores is not supported when flash "
+                "attention is enabled. Please disable flash attention to access"
+                " attention scores."
+            )
+        if config.is_flash_attention_enabled():
+            # Directly compute the attention output using flash attention
+            attention_output = ops.dot_product_attention(
+                query=query,
+                key=key,
+                value=value,
+                mask=attention_mask,
+                scale=self._inverse_sqrt_key_dim,
+                is_causal=bool(self._attention_axes),
+            )
+            # Return only the attention output, as scores are not separately
+            # available
+            return attention_output, None
+
+        # Default behavior without flash attention, with explicit attention
```

Comment:
Can we always go to `dot_product_attention` unless `return_attention_scores` is enabled?